recipes/cpython/all/conanfile.py

```diff
@@ -341,135 +352,110 @@ def project_build(name):
-            })
         return archs
 
     def _msvc_build(self):
         msbuild = MSBuild(self)
-        msbuild_properties = {
-            "IncludeExternals": "true",
-        }
+        msbuild.platform = self._msvc_archs[str(self.settings.arch)]
+
         projects = self._solution_projects
-        self.output.info("Building {} Visual Studio projects: {}".format(len(projects), projects))
+        self.output.info(f"Building {len(projects)} Visual Studio projects: {projects}")
 
-        with tools.no_op():
-            for project_i, project in enumerate(projects, 1):
-                self.output.info("[{}/{}] Building project '{}'...".format(project_i, len(projects), project))
-                project_file = os.path.join(self._source_subfolder, "PCbuild", project + ".vcxproj")
-                self._upgrade_single_project_file(project_file)
-                msbuild.build(project_file, upgrade_project=False, build_type="Debug" if self.settings.build_type == "Debug" else "Release",
-                              platforms=self._msvc_archs, properties=msbuild_properties)
+        sln = os.path.join(self.source_folder, "PCbuild", "pcbuild.sln")
+        # FIXME: Solution files do not pick up the toolset automatically.
+        cmd = msbuild.command(sln, targets=projects)
+        self.run(f"{cmd} /p:PlatformToolset={msvs_toolset(self)}")
 
     def build(self):
-        # FIXME: these checks belong in validate, but the versions of dependencies are not available there yet
-        if self._supports_modules:
-            if tools.Version(self._version_number_only) < "3.8.0":
-                if tools.Version(self.deps_cpp_info["mpdecimal"].version) >= "2.5.0":
-                    raise ConanInvalidConfiguration("cpython versions lesser then 3.8.0 require a mpdecimal lesser then 2.5.0")
-            elif tools.Version(self._version_number_only) >= "3.9.0":
-                if tools.Version(self.deps_cpp_info["mpdecimal"].version) < "2.5.0":
-                    raise ConanInvalidConfiguration("cpython 3.9.0 (and newer) requires (at least) mpdecimal 2.5.0")
-
-        if self._with_libffi:
-            if tools.Version(self.deps_cpp_info["libffi"].version) >= "3.3" and self.settings.compiler == "Visual Studio" and "d" in str(self.settings.compiler.runtime):
-                raise ConanInvalidConfiguration("libffi versions >= 3.3 cause 'read access violations' when using a debug runtime (MTd/MDd)")
-
         self._patch_sources()
-        if self.settings.compiler == "Visual Studio":
+        if is_msvc(self):
             self._msvc_build()
         else:
-            autotools = self._configure_autotools()
+            autotools = Autotools(self)
+            # For debugging configure errors
+            try:
+                autotools.configure()
+            except ConanException:
+                with open(os.path.join(self.build_folder, "config.log"), 'r') as f:
+                    self.output.info(f.read())
+                raise
```

Comment:
```suggestion
            autotools.configure()
```

Let's keep it simple, please. Who is working on the recipe is able to access the build folder and read any file.

Reply:
Fair enough, this was mainly to debug the xvmc issue in CCI.

Reply:
Done